src/components/CytoscapeGraph/Layout/BoxLayout.ts

```diff
@@ -245,25 +207,120 @@ export default class BoxLayout {
         boxNode.removeScratch(PARENT_POSITION_KEY);
       });
     });
+
     layout.run();
   }
 
-  async runDiscrete(layout) {
-    layout.on('layoutstart layoutready layoutstop', _evt => {
-      // Avoid propagating any local layout events up to cy, this would yield a global operation before the nodes are ready.
+  async layoutBoxType(boxByType: BoxByType): Promise<LayoutBoxTypeResult> {
```

Comment:
ídem: minor: since you are not using `await` inside this func, you can remove the `async` keyword.

Reply:
see above comment.